model/registry/hardware.yaml

```diff
@@ -0,0 +1,113 @@
+groups:
+  - id: registry.hardware
+    prefix: hw
+    type: attribute_group
+    brief: >
+      Attributes for hardware.
+    attributes:
+      - id: id
+        type: string
+        stability: experimental
+        brief: >
+          An identifier for the hardware component, unique within the monitored host
+        examples: ['win32battery_battery_testsysa33_1']
+      - id: name
+        type: string
+        stability: experimental
+        brief: >
+          An easily-recognizable name for the hardware component
+        examples: ['eth0']
+      - id: parent
+        type: string
+        stability: experimental
+        brief: >
+          Unique identifier of the parent component (typically the `id` attribute of the enclosure, or disk controller)
+        examples: ['dellStorage_perc_0']
+      - id: type
+        type:
+          allow_custom_values: true
+          members:
+            - id: battery.yaml
+              value: 'battery'
+              brief: "Battery"
+              stability: experimental
+            - id: cpu
+              value: 'cpu'
+              brief: 'CPU'
+              stability: experimental
+            - id: disk_controller
+              value: 'disk_controller'
+              brief: 'Disk controller'
+              stability: experimental
+            - id: enclosure
+              value: 'enclosure'
+              brief: 'Enclosure'
+              stability: experimental
+            - id: fan
+              value: 'fan'
+              brief: 'Fan'
+              stability: experimental
+            - id: gpu
+              value: 'gpu'
+              brief: 'GPU'
+              stability: experimental
+            - id: logical_disk
+              value: 'logical_disk'
+              brief: 'Logical disk'
+              stability: experimental
+            - id: memory
+              value: 'memory'
+              brief: 'Memory'
+              stability: experimental
+            - id: network
+              value: 'network'
+              brief: 'Network'
+              stability: experimental
+            - id: physical_disk
+              value: 'physical_disk'
+              brief: 'Physical disk'
+              stability: experimental
+            - id: power_supply
+              value: 'power_supply'
+              brief: 'Power supply'
+              stability: experimental
+            - id: tape_drive
+              value: 'tape_drive'
+              brief: 'Tape drive'
+              stability: experimental
+            - id: temperature
+              value: 'temperature'
+              brief: 'Temperature'
+              stability: experimental
+            - id: voltage
+              value: 'voltage'
+              brief: 'Voltage'
+              stability: experimental
+        stability: experimental
+        brief: >
+          Type of the component
+      - id: error.type
```

Comment:
there is no need to declare `hw.error.type`, we should reuse the existing one

Reply:
this error type has specific enum value (will be updated in the future PRs). I can't redefine enum values from existing one in current build system.

Reply:
We need extensible enums, because the semantic conventions cannot cover all possible values of error types.

Reply:
@bertysentry we have discussed extensible enums already, maybe they will be there in the near future 
cc @jsuereth @lquerel 
But as for now I will just redefine enum with all possible options as soon as I will move them over

Reply:
Alright, thank you!

Reply:
error.type is intended to be common attribute describing an error, individual semconv can define additional domain specifc attributes, or specify what `error.type` means in their context. 

Let's reuse the generic one as the first step.

Reply:
@lmolkova The intent of this PR is just to migrate from Markdown to YAML. Semantic changes must be brought in a separate PR.

Reply:
@lmolkova I would be ok to use existing `error.type` but I don't see how I can redefine enum unless I redefine the global one.

I would need to make `error.type` an enum of fields such as for example `zero_buffer_credit`, `crc`, `corrected` etc depending on the type of hardware. There is no reusable enums currently. So the only thing I could do here would be update comments and put examples there. 

@bertysentry is it ok from your perspective if type of error is not defined as enum but could have any value? We do not restrict enums currently anyway but sometimes it is nice to have predefined set of values. Here I can put set of values only to examples session